pp_hot.c

```diff
@@ -3205,30 +3205,60 @@ PP(pp_match)
     /* push captures on stack */
 
     {
-        const I32 nparens = RXp_NPARENS(prog);
-        I32 i = (global && !nparens) ? 1 : 0;
+        const I32 logical_nparens = RXp_LOGICAL_NPARENS(prog);
+        I32 logical_paren = (global && !logical_nparens) ? 1 : 0;
+        I32 *l2p = RXp_LOGICAL_TO_PARNO(prog);
+        I32 *p2l_next = RXp_PARNO_TO_LOGICAL_NEXT(prog);
+        SSize_t offs_start, offs_end;
```

Comment:
These are only used inside the while loop and so the definitions could be moved there.

Reply:
willdo.